04-cs-fundamentals/classroom/Sorting.md

```diff
@@ -177,14 +254,81 @@ Overall, each level takes *O(n)* time. There are *O(log n)* such levels. Resulti
 
 Note: We will look closer at merge sort and its implementation when we discuss *Recursive algorithms* in the future.
 
-#### Concluding Notes
-- There are several sorting algorithms available. Whilst we reviewed four of them, you can learn more about the rest on [geeksforgeeks.org/sorting-algorithms/](http://www.geeksforgeeks.org/sorting-algorithms/)
+For small arrays insertion sort performs better than merge sort, but as the size of the array grows merge sort becomes much more performant.
+
+## Comparing Things Beyond Numbers
+
+Often we need to sort items which are not numeric.  How does one sort Student objects, or Slack Channels?  The answer is Ruby's Spaceship Operator `<=>`!
+
+The spaceship operator compares two objects `a <=> b` and returns
+
+- -1 if a < b
+- 0  if a == b
+- 1  if a > b
+
+When you make a class you can override the `def <=> (b)` method provided by Ruby's object class.  In the class below we overrode the inherited method.  If a student is being compared with a non-student object the method will use the student name and the String class' implementation of `<=>`.  If the other object is an instance of Student, then the method will return -1 if the current student's name comes earlier than `other_value`.  If they have the same name then the method will return 0 and if the current student comes later it will return 1.
+
+```ruby
+class Student
+  attr_reader :name, :grade
+
+  def initialize(name, grade)
+    @name = name
+    @grade = grade
+  end
+
+  def <=> (other_value)
```

Comment:
I don't love that this example just uses `name`, I'd like to see it use `grade` as a tiebreaker or something, just to demonstrate that.  (Since as is that bottom portion can just be `name <=> other_value.name`.

Reply:
Added example of using grade